### DISCUSSION TITLE
Fundamentos Do EntityFramework - Erro inserir um Post

### DISCUSSION THREAD

**Discussion by RaMadaSilva** [balta-io/forum#92] [Tira Dúvidas] [answered]
Segui todos os passas da aulas fiz as relações mas no momento de inserir um  novo post no banco de dados  me aparece a seguinte mensagem de erro: 
nhandled exception. Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot insert the value NULL into column 'AuthorId', table 'Blog.dbo.Post'; column does not allow nulls. INSERT fails.
The statement has been terminated.

Eis o código da classe Program: 
  var user = context.Users.FirstOrDefault(x => x.Id == 1004);
            var post = new Post()
            {
                Title = "Aprendendo C# Na pratica",
                Summary = "C#",
                Body = "aprender C# na pratica é um trabalho que exige muita repetição e muita escrita de codigo",
                Slug = "pratic",
                Category = new Category()
                {
                    Name = "Desenvolvimento Backend",
                    Slug = "Des_Backend"
                },
                Author = user
            };

            context.Posts.Add(post);
            context.SaveChanges();

Eis o Codigo da modelo do post: 

namespace FluentBlog.Models
{
    public class Post
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Summary { get; set; }
        public string Body { get; set; }
        public string Slug { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime LastUpdate { get; set; }
        public Category Category { get; set; }
        public User Author { get; set; }

        public List<Tag> Tags { get; set; }

        public Post()
        {
            Tags = new List<Tag>();
        }
    }

Eis o codigo do PostMap: 

using System;
using System.Collections.Generic;
using FluentBlog.Models;
using Microsoft.EntityFrameworkCore;

namespace FluentBlog.DataBase.Mapping
{
    public class PostMap : IEntityTypeConfiguration<Post>
    {
        public void Configure(Microsoft.EntityFrameworkCore.Metadata.Builders.EntityTypeBuilder<Post> builder)
        {
            builder.ToTable("Post");

            builder.HasKey(x => x.Id);

            builder.Property(x => x.Id)
                .ValueGeneratedOnAdd()
                .UseIdentityColumn();

            builder.Property(x => x.Title)
                .IsRequired()
                .HasColumnName("Title")
                .HasColumnType("VARCHAR")
                .HasMaxLength(160);

            builder.Property(x => x.Summary)
                .IsRequired()
                .HasColumnName("Summary")
                .HasColumnType("VARCHAR")
                .HasMaxLength(255);

            builder.Property(x => x.Body)
                .IsRequired()
                .HasColumnName("Body")
                .HasColumnType("TEXT");

            builder.Property(x => x.Slug)
                .IsRequired()
                .HasColumnName("Slug")
                .HasColumnType("VARCHAR")
                .HasMaxLength(80);

            builder.Property(x => x.CreateDate)
                .IsRequired()
                .HasColumnName("CreateDate")
                .HasColumnType("DATETIME")
                .HasDefaultValueSql("GETDATE()");

            builder.Property(x => x.LastUpdate)
                .IsRequired()
                .HasColumnName("LastUpdate")
                .HasColumnType("DATETIME")
                .HasDefaultValueSql("GETDATE()");

            //Relacionamento
            builder.HasOne(x => x.Author)
                .WithMany(x => x.Posts)
                .HasConstraintName("FK_Post_Author")
                .OnDelete(DeleteBehavior.Cascade);

            builder.HasOne(x => x.Category)
                .WithMany(x => x.Posts)
                .HasConstraintName("FK_Post_Category")
                .OnDelete(DeleteBehavior.Cascade);

            builder.HasMany(x => x.Tags)
                .WithMany(x => x.Posts)
                .UsingEntity<Dictionary<string, Object>>(
                    "PostTag", post => post.HasOne<Tag>()
                                    .WithMany()
                                    .HasForeignKey("PostId")
                                    .HasConstraintName("FK_PostTag_PostId")
                                    .OnDelete(DeleteBehavior.Cascade),
                                tag => tag.HasOne<Post>()
                                    .WithMany()
                                    .HasForeignKey("TagId")
                                    .HasConstraintName("FK_PostTag_TagId")
                                    .OnDelete(DeleteBehavior.Cascade));

            builder.HasIndex(x => x.Slug, "IX_Slug_Post")
                .IsUnique();
        }
    }
}

**Comment by RaMadaSilva** [comment]
@andrebaltieri 
consegui resolver o problema que tive acrescentando uma propriedade na classe do modelo.
mas acredito que essa não seja a melhor solução. 

desta feita a classe de modelo do Post ficou da seguinte forma: 
![image](https://user-images.githubusercontent.com/91338367/200814839-16c3606a-914c-4abd-851e-2475bfeb983d.png)

**Comment by Romulo-Queiroz** [comment, marked as answer]
Poderia enviar uma foto da sua classe Post?
